### PR TITLE
[14.0][FIX] fastapi

### PR DESCRIPTION
**Pull request by xavier-bouquiaux**
fastapi_endpoint.py -> the url is relative so it does not work via the interface <br/>
~common.py -> odoo14 <> odoo16 (transaction vs cursor) for full details see sbi~

- [x] depends on https://github.com/OCA/server-tools/pull/2666

### CODE REVIEW

File: fastapi/models/fastapi_endpoint.py [hunk 1]
```diff
@@ -80,10 +80,11 @@ def _check_root_path(self):
...
     @api.depends("root_path")
     def _compute_urls(self):
+        base_url = self.env["ir.config_parameter"].sudo().get_param("web.base.url")
         for rec in self:
-            rec.docs_url = f"{rec.root_path}/docs"
-            rec.redoc_url = f"{rec.root_path}/redoc"
-            rec.openapi_url = f"{rec.root_path}/openapi.json"
+            rec.docs_url = f"{base_url}{rec.root_path}/docs"
+            rec.redoc_url = f"{base_url}{rec.root_path}/redoc"
+            rec.openapi_url = f"{base_url}{rec.root_path}/openapi.json"
```
Comment (lmignon) [inline fastapi/models/fastapi_endpoint.py:98]: Vetry strange since I tested the UI on V14 without these changes and everything is working.

Comment (lmignon) [reply fastapi/models/fastapi_endpoint.py:98]: Except that I added the base_url by hand ;-)

**Comment by lmignon** [comment]
@xavier-bouquiaux Thank you for the PR. All the proposed changes make sense. Nevertheless, can you split the two commits plz? 
For the commit where you modify the odoo environement when running the tests, can you remove `for full details see sbi' from the commit message?

**Comment by sbidoul** [comment]
@lmignon @xavier-bouquiaux I updated this PR to move the contextvars patch to a `base_contextvars` module (https://github.com/OCA/server-tools/pull/2666).

**Comment by sbidoul** [comment]
@xavier-bouquiaux my last change here should fix the issues you had when running with `workers=0`.

**Review by lmignon — APPROVED** [review]
Thank you @sbidoul for the fix. LGTM (Code review)

**Comment by xavier-bouquiaux** [comment]
Thanks @sbidoul and @lmignon 

I've just rebased this on 14.0 as the depends has been merged

Let me know if it's ok or if i still need to do something

regards

xbo

**Comment by lmignon** [comment]
@xavier-bouquiaux Can you confirm that this PR fix your issue with worker=0?

**Review by lmignon — APPROVED** [review]

**Pull request merged**